core/common/src/main/java/alluxio/collections/IndexDefinition.java

```diff
@@ -14,17 +14,26 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * A class representing the definition of an index for this {@link IndexedSet}. Each instance of
- * this class must implement the method to define how to get the value of the field chosen as
- * the index key. Users use this indexDefinition class as the parameter in all methods of
- * {@link IndexedSet} to represent an index.
+ * A class representing an index for an {@link IndexedSet}.
  *
- * @param <T> type of objects in this {@link IndexedSet}
+ * For example, if an indexed set stores inodes, and we want to be able to look them up by their
+ * "id" field, we would define an index like so:
+ *
+ * <pre>
+ * public class IndexDefinition<Inode, Long>() {
```

Comment:
discussion: is it better to make `IndexDefinition` take a functor (`T->V`) in constructor, rather than implementing the  `getFieldValue` abstract method? I believe the code will be shorter when defining these indexes.

Reply:
Let's get github issues going and create a help-wanted task for this

Reply:
That sounds better. This was the way to do it back in java 7 days.